Rover/Map.cpp

```diff
@@ -0,0 +1,327 @@
+#include "Map.h"
+#include <queue>
+#include <cmath>
+#include <algorithm>
+
+void RoverPathfinding::Map::AddObstacle(point coord1, point coord2)
+{
+    obstacle o;
+    o.marked = false;
+    o.coord1 = coord1;
+    o.coord2 = coord2;
+    obstacles.push_back(o);
+}
+
+// given lat, long, bearing (in degrees), and distance (in km), returns a new point
+RoverPathfinding::point RoverPathfinding::Map::lat_long_offset(float lat1, float lon1, float brng, float dist)
+{
+    dist /= 1000.0f;
+#define PI 3.14159265359
+#define R_EARTH 6371.0088 // in km
+    lat1 = lat1 * (PI / 180); // into radians
+    lon1 = lon1 * (PI / 180); // rad
+    brng = brng * (PI / 180); // rad    
+    float lat2 = asin(sin(lat1) * cos(dist / R_EARTH) + cos(lat1) * sin(dist / R_EARTH) * cos(brng));
+    float lon2 = lon1 + atan2(sin(brng) * sin(dist / R_EARTH) * cos(lat1), cos(dist / R_EARTH) - sin(lat1) * sin(lat2));
+    lat2 = lat2 * (180 / PI); // into degrees
+    lon2 = lon2 * (180 / PI); // deg
+#undef PI
+#undef R_EARTH
+    return(std::make_pair(lat2, lon2));
+
+}
+
+//start, end, circle, and R are in lat/long coordinates
+bool RoverPathfinding::Map::segment_circle_intersection(point start,
```

Comment:
How about segment_circle_intersects as a name? Intersection sounds like it returns some information about the intersection.